src/Compilers/Core/Portable/CommandLine/CommonCompiler.cs

```diff
@@ -547,7 +579,7 @@ internal static bool HasUnsuppressedErrors(DiagnosticBag diagnostics)
         {
             foreach (var diag in diagnostics.AsEnumerable())
             {
-                if (IsReportedError(diag))
+                if (diag.DefaultSeverity == DiagnosticSeverity.Error && !diag.IsSuppressed)
```

Comment:
As I mentioned offline, I'd like to see this change and the tests for it in a separate PR.

Reply:
I included in this PR as it was a pretty simple change that @AlekseyTs proposed. I can move to a separate PR if there is specific concern on having it part of this PR.

Reply:
I prefer all logically separate changes to be in different PRs, especially if the diff has thousands of lines of changes and tens of files changed.

Reply:
Without this change the feature would work incorrectly in presence of /warnaserror. Additionally, it is a simple one line change for which unit tests have been added. IMO there is no value add in separating it out, but I am fine to do so given this is compiler team's code.

Reply:
I don't know where you are seeing "one line changed" in this commit, https://github.com/dotnet/roslyn/pull/36067/commits/d55ad54453d58fb2a978323d1844db4239fc3209 but something on your Github diff is seriously broken :)

Reply:
Majority of those changes in that commit were revert of the functionality added in a prior commit, based on the new approach suggested by @AlekseyTs

Reply:
Reverted /warnaserror scenario changes with https://github.com/dotnet/roslyn/pull/36067/commits/56c9710f788fb3a818d0f993268cb1eeb13d8bf7.

Filed https://github.com/dotnet/roslyn/issues/36215 to track reverting the product changes in this commit and unskipping tests for the /warnaserror scenario to work fine again with a separate PR.